### DISCUSSION TITLE
TTL delete clause referring another table

### DISCUSSION THREAD

**Discussion by MaheshGPai** [ClickHouse/ClickHouse#53911] [Q&A] [answered]
Does clickhouse support TTL delete clause that refers another table like the example below

> CREATE TABLE eventTest
> (
>     `eventId` String,
>     `value` UInt64,
>     `time` DateTime
> )
> ENGINE = MergeTree
> ORDER BY (eventId, time)
> TTL time + INTERVAL 1 MINUTE DELETE WHERE eventId not in (select eventId from interestingEvents)

When I tried this creating this table, I did not get any error. But the records did not get deleted. I even tried forcing the TTL byt running the below command as well, but it did not help

> ALTER TABLE default.eventTest MATERIALIZE TTL

Is this supported? If so, what am I missing?

Thanks!

>

**Comment by arisarose** [comment, marked as answer]
@MaheshGPai I checked with our engineers and it was mentioned that any filter applied in the TTL expression must be deterministic.
Similarly, it is not possible to utilize dictGet() in the WHERE clause of a TTL expression.
relevant issue: https://github.com/ClickHouse/ClickHouse/issues/30496

Hope this helps and thanks for reaching out!

**Reply by MaheshGPai** [reply]
Thanks @arisarose for checking!
